src/components/modal/modal.js

```diff
@@ -1039,7 +1039,8 @@ Modal.prototype = {
       self.changeObserver.observe(self.element[0], { childList: true, subtree: true });
       self.setFocusableElems();
 
-      let focusElem = $(self.focusableElems).not('.modal-header .searchfield').first();
+      let focusElem = $(self.focusableElems).not('.modal-header .searchfield')
+        .not('[style*="display: none"]').first();
```

Comment:
```suggestion
        .not(':hidden').first();
```

Will this work? It could catch more cases like it is invisible with css or css visibility. Lets try to see if it solves?